File: langgraph/data/equiv/56.py
def validate_bracket_sequence(sequence: str):
    level = 0
    index = 0
    while index < len(sequence):
        if sequence[index] == "<":
            level += 1
        else:
            level -= 1
        if level < 0:
            return False
        index += 1
    return level == 0

# method name: correct_bracketing -> validate_bracket_sequence
# parameter name: brackets -> sequence
# variable name: depth -> level, b -> (removed, using indexing instead)
# changed for loop to while loop with index
# kept the same logic and return conditions

File: langgraph/result/qwen/14b-eq2js/56.js
function validateBracketSequence(sequence) {
    let level = 0;
    let index = 0;
    while (index < sequence.length) {
        if (sequence[index] === "<") {
            level += 1;
        } else {
            level -= 1;
        }
        if (level < 0) {
            return false;
        }
        index += 1;
    }
    return level === 0;
}